Python/pystate.c

```diff
@@ -92,6 +92,11 @@ _PyRuntimeState_Fini(_PyRuntimeState *runtime)
         runtime->interpreters.mutex = NULL;
     }
 
+    if (runtime->xidregistry.mutex != NULL) {
+        PyThread_free_lock(runtime->xidregistry.mutex);
+        runtime->xidregistry.mutex = NULL;
```

Comment:
Please write a separated PR to this change. I don't know well this code, so I'm not sure how xidregistry is supposed to be used.